python/pyarrow/_compute.pyx

```diff
@@ -569,6 +543,10 @@ cdef class FunctionOptions(_Weakrefable):
         return self.get_options().Equals(deref(other.get_options()))
 
 
+def _raise_invalid_function_option(value, description, *, error="keyword"):
```

Comment:
This will give e.g.:
```"foo" is not a valid count mode keyword```
IMHO it would read better as:
```"foo" is not a valid count mode value```

Reply:
Or could also leave out the final word entirely: `"foo" is not a valid count mode`